onecodex/models/analysis.py

```diff
@@ -40,6 +51,51 @@ class Classifications(Analyses):
     _resource_path = "/api/v1/classifications"
     _cached_table = None
 
+    @classmethod
+    def _append_abundance_rollups(cls, results):
+        """Append cumulative abundances into the results returned from the API.
+        """
+        table = {t["tax_id"]: t.copy() for t in results}
+        for tax_id in table:
+            table[tax_id][Metric.AbundanceWChildren.value] = 0.0
+
+        # Older results might have a bug where the parent is missing,
+        # so we need to zero-out those results and renormalize the data.
+        renormalize = False
+        for tax_id, result in table.items():
+            table[tax_id] = result
+            if result[Metric.Abundance] is not None and result["parent_tax_id"] not in table:
+                renormalize = True
+                table[tax_id][Metric.Abundance] = None
+
+        if renormalize:
+            warnings.warn(
+                "Taxa with an abundance metric but no assigned reads have been removed. In order to avoid this, re-run your samples on the latest One Codex Database."
+            )
+            abundance_sum = sum([t.get(Metric.Abundance, 0) or 0 for tax_id, t in table.items()])
+            for tax_id, result in table.items():
+                if Metric.Abundance in result and result[Metric.Abundance] is not None:
+                    result[Metric.Abundance] = result[Metric.Abundance] / abundance_sum
+
+        # Roll-up abundances to parent taxa
+        for tax_id, result in table.items():
+            if result["parent_tax_id"] not in table:
+                continue
+
+            parent = table[result["parent_tax_id"]]
+            result[Metric.AbundanceWChildren.value] += result[Metric.Abundance] or 0.0
+
+            while parent:
+                parent[Metric.AbundanceWChildren.value] += result[Metric.Abundance] or 0.0
```

Comment:
I'm not sure you need all the `.value` in that function

Reply:
We do need it or the key becomes the enum, which is a bit gross as it's inconsistent with the other keys.